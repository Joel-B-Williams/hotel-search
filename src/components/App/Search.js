import React, { Component } from 'react';

class Search extends Component {

  onChange  = event => {
    this.props.handleChange(event.target.value)
  }

  render() {
    return(
        <div className="filters">
            <label htmlFor="propertySearch">Search properties</label>
            <input 
              type="text" 
              onKeyUp={this.onChange} 
              id="propertySearch"
              placeholder="Property name"/>
            <button onClick={this.props.handleClick} className="btn">Price Low-to-High</button>
        </div>
    )
  }
}

export default Search;