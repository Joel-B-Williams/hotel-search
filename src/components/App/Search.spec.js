import React from 'react';
import { shallow } from 'enzyme';
import Search from './Search';
import { create } from 'react-test-renderer';

describe('Search', () => {
    const wrapper = shallow(<Search />);

    it('renders the component', () => {
      expect(wrapper.find('.filters').exists()).toBe(true);
    });

    it('matches snapshot', () => {
      const comp = create(<Search  />);
      
      expect(comp.toJSON()).toMatchSnapshot();
    });
});