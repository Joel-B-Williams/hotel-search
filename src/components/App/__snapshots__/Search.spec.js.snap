// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search matches snapshot 1`] = `
<div
  className="filters"
>
  <label
    htmlFor="propertySearch"
  >
    Search properties
  </label>
  <input
    id="propertySearch"
    onKeyUp={[Function]}
    placeholder="Property name"
    type="text"
  />
  <button
    className="btn"
  >
    Price Low-to-High
  </button>
</div>
`;
